app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,60 @@
+class OrdersController < ApplicationController
+  include OrdersHelper
+
+  before_action :authenticate_user!, only: :add_to_cart
+
+  def authenticate_user!
+    unless current_user
+      redirect_to login_path, alert: 'You need to log in to add products to the cart'
+    end
+  end
+
+  def add_to_cart
+    filter_params = {
+      category: params[:category],
+      vegetarian: params[:vegetarian],
+      'price-sorting': params['price-sorting'],
+      min_price: params[:min_price],
+      max_price: params[:max_price]
+    }
+
+    # find the product
+    product = Product.find(params[:product_id])
+    # check if the user has an order with status "isCart" true
+    order = Order.find_by(user_id: current_user.id, isCart: true)
+    # if not, create a new order with status "isCart" true, user_id current_user.id and total null as it will be calculated later
+    if order.nil?
+      order = Order.create(user_id: current_user.id, isCart: true)
+    end
```

Comment:
There is a nice feature in Rails called `find_or_create_by(...)` which achieves exactly what you want here.
https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by